CarAccidentErrorHandling/CarAccidentErrorHandling.ts

```diff
@@ -0,0 +1,118 @@
+class SlipperyRoadAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class CollisionAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class HitAndRunAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class EnvironmentalAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+function driveSafely() {
+  const isSlipperyRoad = true;
+  const isCollision = false;
+  const isEnvironmental = false;
+
+  if (isSlipperyRoad) {
+    throw new SlipperyRoadAccidentException(
+      "Slippery road due to rain. Drive with caution."
+    );
+  } else if (isCollision) {
+    throw new CollisionAccidentException(
+      "Collision with another vehicle occurred."
+    );
+  } else if (isEnvironmental) {
+    throw new EnvironmentalAccidentException(
+      "Environmental conditions such as thunderstorm. Drive with caution."
+    );
+  } else {
+    throw new HitAndRunAccidentException("Hit and run incident happened.");
+  }
+}
+
```

Comment:
all these functions below are help in handling the exceptions, so I think it would be better to move it to a separate file.

Reply:
done